go/pkg/client/cas_test.go

```diff
@@ -712,74 +714,71 @@ func TestUpload(t *testing.T) {
 
 	for _, ub := range []client.UseBatchOps{false, true} {
 		for _, uo := range []client.UnifiedCASOps{false, true} {
-			t.Run(fmt.Sprintf("UsingBatch:%t,UnifiedCASOps:%t", ub, uo), func(t *testing.T) {
-				for _, tc := range tests {
-					t.Run(tc.name, func(t *testing.T) {
-						ctx := context.Background()
-						e, cleanup := fakes.NewTestEnv(t)
-						defer cleanup()
-						fake := e.Server.CAS
-						c := e.Client.GrpcClient
-						client.CASConcurrency(defaultCASConcurrency).Apply(c)
-						ub.Apply(c)
-						uo.Apply(c)
-						if tc.concurrency > 0 {
-							tc.concurrency.Apply(c)
-						}
-
-						present := make(map[digest.Digest]bool)
-						for _, blob := range tc.present {
-							fake.Put(blob)
-							present[digest.NewFromBlob(blob)] = true
-						}
-						var input []*uploadinfo.Entry
-						for _, blob := range tc.input {
-							input = append(input, uploadinfo.EntryFromBlob(blob))
-						}
+			for _, cmp := range []client.CompressedBytestreamThreshold{-1, 0} {
```

Comment:
Note to self: add a []client.Opt field to tests and expand the tests to contain the cartesian product of these options in a loop before running them. The indentation is getting out of hand...
(I'll clean this up separately)

Reply:
sgtm - if the indentation is the only problem you can also move the inner block into a separate function. Too late for it right now, but it is useful whereas in it preserves the original blame!